src/portable/microchip/pic32mz/dcd_pic32mz.c

```diff
@@ -163,6 +163,10 @@ void dcd_remote_wakeup(uint8_t rhport)
   USB_REGS->POWERbits.RESUME = 1;
 #if CFG_TUSB_OS != OPT_OS_NONE
   osal_task_delay(10);
+#else
+  // TODO: Wait in non blocking mode
+  unsigned cnt = 2000;
```

Comment:
I am not entirely sure if 2000+ nops is enough for at least 1ms delay. It could have issue with fast mcu. Does pic32 has something like SystemCoreClock. 

```C
  // try to delay for 1 ms
  uint32_t count = SystemCoreClock / 1000;
  while ( count-- )
  {
    __NOP();
  }
```


It is not really a problem for now, we could come back to this later

Reply:
There is no standard SystemCoreClock as provided by CMSIS
There is CPU0 register COUNTER that is kind like Cortex SysTick, I can use it if you insist.
In fact 2000 NOPs at 200MHz is much faster then time you mentioned from datasheet but in my case it was enough to trigger remote wake up every time in contrast to where there is no delay between toggling bit and remote wake up is not activated.

Reply:
if it works, then we will leave it as it is now. And come back with a more accurate delay later on. No problem at all. 

Note since the timing is out of specs, this maybe specific to the host driver (in this case the usbcv), other host driver (linux/windows/macos) may decide this signal is too short to register as remote wakeup.